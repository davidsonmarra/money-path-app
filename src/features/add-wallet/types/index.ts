import {IconType} from 'src/assets/icons/types';

export interface WalletProps {
  name: string;
  icon: IconType;
  color?: string;
  backgroundColor?: string;
  type: 'personal' | 'bank';
  amount?: number;
}

export type SelectWalletSection = {
  title: string;
  data: WalletProps[];
};
