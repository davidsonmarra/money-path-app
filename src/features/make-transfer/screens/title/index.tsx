import React from 'react';
import { useNavigation } from '@react-navigation/native';
import { StackNavigationProp } from '@react-navigation/stack';
import TitleContainer from 'src/features/make-transfer/screens/title/ui';
import { PrivateStackParamList } from 'src/features/navigation';
import useMakeTransferForm from 'src/features/make-transfer/hooks/make-transfer-form';
import { titleSchema } from 'src/features/make-transfer/context/make-transfer-form/schema';

type MakeTransferScreenNavigationProp = StackNavigationProp<
  PrivateStackParamList,
  'MakeTransferStack'
>;

const TitleScreen = () => {
  const { watch, setValue } = useMakeTransferForm();
  const navigation = useNavigation<MakeTransferScreenNavigationProp>();

  const handleOnConfirm = () => {
    navigation.navigate('AddWalletStack', {
      screen: 'InitialValue',
    });
  };

  const handleValidateForm = () => {
    const { success } = titleSchema.safeParse({
      title: watch('title'),
    });
    return success;
  };

  return (
    <TitleContainer
      onBack={navigation.goBack}
      onConfirm={handleOnConfirm}
      setTitle={value => setValue('title', value)}
      title={watch('title')}
      isDisabled={!handleValidateForm()}
    />
  );
};

export default TitleScreen;
