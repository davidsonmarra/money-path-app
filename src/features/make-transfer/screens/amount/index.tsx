import React, { useEffect, useMemo, useState } from 'react';
import { StackNavigationProp } from '@react-navigation/stack';
import { useNavigation } from '@react-navigation/native';
import AmountContainer from 'src/features/make-transfer/screens/amount/ui';
import { PrivateStackParamList } from 'src/features/navigation';
import useMakeTransferForm from '../../hooks/make-transfer-form';
import { IconType } from 'src/assets/icons/types';
import { useWalletStore } from 'src/hooks/useWallet';

type MakeTransferScreenNavigationProp = StackNavigationProp<
  PrivateStackParamList,
  'MakeTransferStack'
>;

export interface WalletUiProps {
  id: string;
  label: string;
  icon: IconType;
  iconBackground: string;
  type: 'bank' | 'wallet';
  balance: number;
  value: string;
}

const AmountScreen = () => {
  const { wallets, getWallets } = useWalletStore();
  const { watch, setValue } = useMakeTransferForm();
  const [sourceWallet, setSourceWallet] = useState<WalletUiProps>();
  const [destinationWallet, setDestinationWallet] = useState<WalletUiProps>();
  const [showDatePicker, setShowDatePicker] = useState(false);
  const [tempDate, setTempDate] = useState<Date | null>(null);
  const navigation = useNavigation<MakeTransferScreenNavigationProp>();

  const handleOnBack = () => {
    navigation.goBack();
  };

  const handleOnConfirm = () => {
    navigation.navigate('MakeTransferStack', {
      screen: 'Title',
    });
  };

  const handleChangeText = (text: string) => {
    const numericValue = text.replace(/\D/g, '');
    setValue('amount', Number(numericValue));
  };

  const handleSelectSourceWallet = (walletId: string) => {
    const wallet = formattedWallets.find(wallet => wallet.id === walletId);
    setSourceWallet(wallet);
    setValue('sourceWalletId', walletId);
  };

  const handleSelectDestinationWallet = (walletId: string) => {
    const wallet = formattedWallets.find(wallet => wallet.id === walletId);
    setDestinationWallet(wallet);
    setValue('destinationWalletId', walletId);
  };

  const handleSwapWallets = () => {
    const temp = sourceWallet;
    setSourceWallet(destinationWallet);
    setDestinationWallet(temp);
  };

  const handleDateChange = (date: Date) => {
    setTempDate(date);
  };

  const handleOpenDatePicker = () => {
    const currentTimestamp = watch('timestamp');
    setTempDate(currentTimestamp || new Date());
    setShowDatePicker(true);
  };

  const handleConfirmDate = () => {
    if (tempDate) {
      setValue('timestamp', tempDate);
      setShowDatePicker(false);
    }
  };

  const handleCancelDate = () => {
    setTempDate(null);
    setShowDatePicker(false);
  };

  const formatValue = (rawValue: string): string | undefined => {
    if (!rawValue) return undefined;

    const numericValue = parseInt(rawValue.replace(/\D/g, ''), 10);
    return (numericValue / 100).toFixed(2).replace('.', ',');
  };

  const formattedWallets: WalletUiProps[] = useMemo(() => {
    return wallets.map(wallet => ({
      id: wallet.id,
      label: wallet.name,
      icon: wallet.icon,
      iconBackground: wallet.backgroundColor,
      type: 'wallet',
      balance: wallet.balance,
      value: wallet.id,
    }));
  }, [wallets]);

  useEffect(() => {
    getWallets();
  }, []);

  useEffect(() => {
    setSourceWallet(formattedWallets[0]);

    if (watch('type') === 'transfer') {
      setDestinationWallet(formattedWallets[1]);
    }
  }, [wallets]);

  useEffect(() => {
    if (!watch('timestamp')) {
      setValue('timestamp', new Date());
    }
  }, []);

  return (
    <AmountContainer
      onBack={handleOnBack}
      onConfirm={handleOnConfirm}
      amount={formatValue(watch('amount')?.toString() ?? '') ?? '0,00'}
      transferBeetweenWallets={watch('type') === 'transfer'}
      onChangeAmount={handleChangeText}
      wallets={formattedWallets.filter(
        wallet =>
          wallet.id !== sourceWallet?.id && wallet.id !== destinationWallet?.id,
      )}
      selectedSourceWallet={sourceWallet}
      selectedDestinationWallet={destinationWallet}
      onSelectSourceWallet={handleSelectSourceWallet}
      onSelectDestinationWallet={handleSelectDestinationWallet}
      onSwapWallets={handleSwapWallets}
      timestamp={watch('timestamp')}
      showDatePicker={showDatePicker}
      onDateChange={handleDateChange}
      onOpenDatePicker={handleOpenDatePicker}
      onCloseDatePicker={handleCancelDate}
      onConfirmDate={handleConfirmDate}
      tempDate={tempDate}
    />
  );
};

export default AmountScreen;
