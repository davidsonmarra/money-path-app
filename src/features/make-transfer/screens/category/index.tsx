import React, { useCallback, useEffect, useState } from 'react';
import { useNavigation } from '@react-navigation/native';
import { StackNavigationProp } from '@react-navigation/stack';
import CategoryContainer from 'src/features/make-transfer/screens/category/ui';
import { PrivateStackParamList } from 'src/features/navigation';
import useMakeTransferForm from 'src/features/make-transfer/hooks/make-transfer-form';
import { IconType } from 'src/assets/icons/types';
import { useCategoriesStore } from 'src/hooks/useCategories';

type MakeTransferScreenNavigationProp = StackNavigationProp<
  PrivateStackParamList,
  'MakeTransferStack'
>;

const mockCategories = [
  {
    id: '1',
    name: 'Alimentação',
    backgroundColor: '#FF6B35',
    color: '#FFFFFF',
    icon: IconType.food,
  },
  {
    id: '2',
    name: 'Transporte',
    backgroundColor: '#4A90E2',
    color: '#FFFFFF',
    icon: IconType.transport,
  },
  {
    id: '3',
    name: 'Casa',
    backgroundColor: '#8B4513',
    color: '#FFFFFF',
    icon: IconType.house,
  },
  {
    id: '4',
    name: 'Lazer',
    backgroundColor: '#9C27B0',
    color: '#FFFFFF',
    icon: IconType.games,
  },
  {
    id: '5',
    name: 'Saúde',
    backgroundColor: '#E91E63',
    color: '#FFFFFF',
    icon: IconType.health,
  },
  {
    id: '6',
    name: 'Outros',
    backgroundColor: '#607D8B',
    color: '#FFFFFF',
    icon: IconType.help,
  },
];

export type Status = 'loading' | 'default' | 'error';

interface Props {
  initialStatus?: Status;
}

const CategoryScreen = ({ initialStatus = 'loading' }: Props) => {
  const { watch, setValue } = useMakeTransferForm();
  const navigation = useNavigation<MakeTransferScreenNavigationProp>();
  const [status, setStatus] = useState<Status>(initialStatus);

  const { categories, createCategory } = useCategoriesStore();

  const initializeCategories = async () => {
    try {
      for (const category of mockCategories) {
        await createCategory(category);
        await new Promise(resolve => setTimeout(resolve, 200));
      }
    } catch (error) {
      setStatus('error');
    } finally {
      setStatus('default');
    }
  };

  useEffect(() => {
    if (status === 'loading') {
      initializeCategories();
    }
  }, []);

  const handleOnConfirm = () => {
    navigation.navigate('MakeTransferStack', {
      screen: 'Title',
    });
  };

  return (
    <CategoryContainer
      onBack={navigation.goBack}
      onConfirm={handleOnConfirm}
      setCategory={value => setValue('categoryId', value)}
      selectedCategoryId={watch('categoryId')}
      categories={categories}
      status={status}
    />
  );
};

export default CategoryScreen;
