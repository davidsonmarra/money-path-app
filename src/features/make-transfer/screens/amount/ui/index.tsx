import React from 'react';
import { TouchableOpacity, View } from 'react-native';
import { SafeAreaView } from 'react-native-safe-area-context';
import DatePicker from 'react-native-date-picker';
import {
  BottomSheet,
  Button,
  ButtonType,
  Header,
  IconRound,
  InputAmount,
  SelectionDropdown,
  Spacer,
  Text,
  TextType,
} from 'src/components';
import useStyles from 'src/features/make-transfer/screens/amount/ui/styles';
import { WalletUiProps } from 'src/features/make-transfer/screens/amount';
import { SelectionDropdownItem } from 'src/components/selection-dropdown';
import { IconType } from 'src/assets/icons/types';

interface Props {
  onBack: () => void;
  onConfirm: () => void;
  amount: string;
  onChangeAmount: (value: string) => void;
  transferBeetweenWallets: boolean;
  wallets?: WalletUiProps[];
  selectedSourceWallet?: WalletUiProps;
  selectedDestinationWallet?: WalletUiProps;
  onSelectSourceWallet: (walletId: string) => void;
  onSelectDestinationWallet: (walletId: string) => void;
  onSwapWallets: () => void;
  timestamp?: Date;
  showDatePicker: boolean;
  onDateChange: (date: Date) => void;
  onOpenDatePicker: () => void;
  onCloseDatePicker: () => void;
  onConfirmDate: () => void;
  tempDate?: Date | null;
}

const AmountContainer = ({
  onBack,
  onConfirm,
  amount,
  onChangeAmount,
  transferBeetweenWallets,
  wallets = [],
  selectedSourceWallet,
  selectedDestinationWallet,
  onSelectSourceWallet,
  onSelectDestinationWallet,
  onSwapWallets,
  timestamp,
  showDatePicker,
  onDateChange,
  onOpenDatePicker,
  onCloseDatePicker,
  onConfirmDate,
  tempDate,
}: Props) => {
  const styles = useStyles();

  return (
    <>
      <Header text="Valor da transferência" onLeftIconPress={onBack} />
      <View style={styles.amountContainer}>
        <InputAmount
          value={amount}
          onChangeText={onChangeAmount}
          placeholder="0,00"
          testID="input-amount"
          keyboardType="numeric"
        />
      </View>
      <SafeAreaView edges={['bottom']} style={styles.container}>
        <View>
          {transferBeetweenWallets ? (
            <View>
              <Text type={TextType.headingMedium}>Selecione as carteiras</Text>
              <Spacer height={12} />
              <Text type={TextType.textMediumRegular}>
                Para criar uma transferência, precisamos que você informe a
                carteira de origem e destino:
              </Text>
              <Spacer height={16} />
              <View style={styles.walletsContainer}>
                <View>
                  <Text
                    type={TextType.textSmallRegular}
                    style={styles.dropdownLabel}
                  >
                    De
                  </Text>
                  <SelectionDropdown
                    items={wallets}
                    selectedItem={selectedSourceWallet as SelectionDropdownItem}
                    onSelectedItem={item => onSelectSourceWallet?.(item.value)}
                    placeholder="Carteira origem"
                    showIcon
                    testID="source-wallet-dropdown"
                  />
                </View>
                <TouchableOpacity
                  onPress={onSwapWallets}
                  testID="swap-wallets-button"
                >
                  <IconRound icon={IconType.change} size={56} />
                </TouchableOpacity>

                <View>
                  <Text
                    type={TextType.textSmallRegular}
                    style={styles.dropdownLabel}
                  >
                    Para
                  </Text>
                  <SelectionDropdown
                    items={wallets}
                    selectedItem={
                      selectedDestinationWallet as SelectionDropdownItem
                    }
                    onSelectedItem={item =>
                      onSelectDestinationWallet?.(item.value)
                    }
                    placeholder="Carteira destino"
                    showIcon
                    testID="destination-wallet-dropdown"
                  />
                </View>
              </View>
            </View>
          ) : (
            <View>
              <Text type={TextType.headingMedium}>Selecione a carteira</Text>
              <Spacer height={12} />
              <Text type={TextType.textMediumRegular}>
                Para criar uma transferência, precisamos que você informe a
                carteira:
              </Text>
              <Spacer height={12} />
              <View style={styles.walletContainer}>
                <SelectionDropdown
                  items={wallets}
                  selectedItem={selectedSourceWallet as SelectionDropdownItem}
                  onSelectedItem={item => onSelectSourceWallet?.(item.value)}
                  placeholder="Carteira"
                  showIcon
                  testID="source-wallet-dropdown"
                />
              </View>
            </View>
          )}
          <View style={styles.dateContainer}>
            <Text type={TextType.textMediumRegular}>
              Data e hora da transferência
            </Text>
            <Spacer height={8} />
            <TouchableOpacity
              onPress={onOpenDatePicker}
              style={styles.dateButton}
              testID="date-button"
            >
              <Text type={TextType.headingMedium} style={styles.dayText}>
                {timestamp?.toLocaleString('pt-BR', {
                  day: '2-digit',
                })}
              </Text>
              <Text type={TextType.textMediumRegular} style={styles.monthText}>
                de{' '}
                {timestamp?.toLocaleString('pt-BR', {
                  month: 'long',
                })}
              </Text>
              <Spacer height={8} />
              <Text type={TextType.textMediumRegular} style={styles.hourText}>
                {timestamp?.toLocaleString('pt-BR', {
                  hour: '2-digit',
                  minute: '2-digit',
                })}
              </Text>
            </TouchableOpacity>
          </View>
        </View>
        <View>
          <Button
            type={ButtonType.primary}
            text="Continuar"
            onPress={onConfirm}
            testID="btn-confirm"
          />
        </View>
      </SafeAreaView>
      <BottomSheet
        isVisible={showDatePicker}
        onClose={onCloseDatePicker}
        title="Selecionar data e hora"
        maxHeight="100%"
        testID="date-picker-bottom-sheet"
      >
        <View style={styles.datePickerContainer}>
          <DatePicker
            date={tempDate || timestamp || new Date()}
            onDateChange={onDateChange}
            mode="date"
          />
        </View>
        <Spacer height={16} />
        <View style={styles.modalButtons}>
          <TouchableOpacity
            onPress={onCloseDatePicker}
            style={styles.cancelButton}
          >
            <Text type={TextType.textMediumRegular}>Cancelar</Text>
          </TouchableOpacity>
          <TouchableOpacity
            onPress={onConfirmDate}
            style={[
              styles.confirmButton,
              !tempDate && styles.confirmButtonDisabled,
            ]}
            disabled={!tempDate}
          >
            <Text
              type={TextType.textMediumRegular}
              style={styles.confirmButtonText}
            >
              Confirmar
            </Text>
          </TouchableOpacity>
        </View>
      </BottomSheet>
    </>
  );
};

export default AmountContainer;
