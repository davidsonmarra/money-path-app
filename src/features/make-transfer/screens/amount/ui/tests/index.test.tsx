import React from 'react';
import { fireEvent, render } from 'src/configs/test-utils';
import AmountContainer from 'src/features/make-transfer/screens/amount/ui';
import { WalletUiProps } from 'src/features/make-transfer/screens/amount';
import { IconType } from 'src/assets/icons/types';

// Mock react-native-modal
jest.mock('react-native-modal', () => {
  const React = require('react');
  const { View } = require('react-native');

  return function MockModal({
    isVisible,
    children,
    onBackdropPress,
    onBackButtonPress,
    ...props
  }: any) {
    if (!isVisible) return null;

    return React.createElement(
      View,
      {
        testID: 'modal',
        onPress: onBackdropPress,
        ...props,
      },
      children,
    );
  };
});

// Mock react-native-date-picker
jest.mock('react-native-date-picker', () => {
  const React = require('react');
  const { View, Text } = require('react-native');

  return function MockDatePicker({ date, onDateChange, ...props }: any) {
    return React.createElement(
      View,
      {
        testID: 'date-picker',
        onPress: () => onDateChange && onDateChange(new Date()),
        ...props,
      },
      React.createElement(Text, {}, 'DatePicker Mock'),
    );
  };
});

// Mock BottomSheet component
jest.mock('src/components/bottom-sheet', () => {
  const React = require('react');
  const { View, Text } = require('react-native');

  return function MockBottomSheet({
    isVisible,
    onClose,
    title,
    children,
    testID,
    ...props
  }: any) {
    if (!isVisible) return null;

    return React.createElement(
      View,
      {
        testID: testID || 'bottom-sheet',
        onClose: onClose,
        ...props,
      },
      [React.createElement(Text, { key: 'title' }, title), children],
    );
  };
});

const mockWallets: WalletUiProps[] = [
  {
    id: '1',
    label: 'Carteira 1',
    icon: IconType.wallet,
    iconBackground: '#FF0000',
    type: 'wallet',
    balance: 1000,
    value: '1',
  },
  {
    id: '2',
    label: 'Carteira 2',
    icon: IconType.bank,
    iconBackground: '#00FF00',
    type: 'bank',
    balance: 2000,
    value: '2',
  },
];

const mockOnBack = jest.fn();
const mockOnConfirm = jest.fn();
const mockOnChangeAmount = jest.fn();
const mockOnSelectSourceWallet = jest.fn();
const mockOnSelectDestinationWallet = jest.fn();
const mockOnSwapWallets = jest.fn();
const mockOnDateChange = jest.fn();
const mockOnOpenDatePicker = jest.fn();
const mockOnCloseDatePicker = jest.fn();
const mockOnConfirmDate = jest.fn();

const containerInstance = (props = {}) =>
  render(
    <AmountContainer
      onBack={mockOnBack}
      onConfirm={mockOnConfirm}
      amount="0,00"
      onChangeAmount={mockOnChangeAmount}
      transferBeetweenWallets={false}
      wallets={mockWallets}
      selectedSourceWallet={mockWallets[0]}
      selectedDestinationWallet={mockWallets[1]}
      onSelectSourceWallet={mockOnSelectSourceWallet}
      onSelectDestinationWallet={mockOnSelectDestinationWallet}
      onSwapWallets={mockOnSwapWallets}
      timestamp={new Date()}
      showDatePicker={false}
      onDateChange={mockOnDateChange}
      onOpenDatePicker={mockOnOpenDatePicker}
      onCloseDatePicker={mockOnCloseDatePicker}
      onConfirmDate={mockOnConfirmDate}
      tempDate={null}
      {...props}
    />,
  );

describe('AmountContainer', () => {
  beforeEach(() => {
    jest.clearAllMocks();
  });

  it('should render correctly', () => {
    const container = containerInstance({}).toJSON();
    expect(container).toMatchSnapshot();
  });

  it('should call onBack when back button is pressed', () => {
    const container = containerInstance({});
    container.getByTestId('btn-left-icon').props.onClick();
    expect(mockOnBack).toHaveBeenCalledTimes(1);
  });

  it('should call onChangeAmount when amount input changes', () => {
    const container = containerInstance({});
    const input = container.getByTestId('input-amount');
    fireEvent.changeText(input, '1000');
    expect(mockOnChangeAmount).toHaveBeenCalledWith('1000');
  });

  it('should call onOpenDatePicker when date button is pressed', () => {
    const container = containerInstance({});
    const dateButton = container.getByTestId('date-button');
    fireEvent.press(dateButton);
    expect(mockOnOpenDatePicker).toHaveBeenCalledTimes(1);
  });

  it('should call onConfirm when confirm button is pressed', () => {
    const container = containerInstance({});
    const button = container.getByTestId('btn-confirm');
    fireEvent.press(button);
    expect(mockOnConfirm).toHaveBeenCalledTimes(1);
  });

  it('should show date picker when showDatePicker is true', () => {
    const container = containerInstance({ showDatePicker: true });
    expect(container.getByTestId('date-picker-bottom-sheet')).toBeTruthy();
    expect(container.getByTestId('date-picker')).toBeTruthy();
  });

  it('should call onCloseDatePicker when date picker is closed', () => {
    const container = containerInstance({ showDatePicker: true });
    const bottomSheet = container.getByTestId('date-picker-bottom-sheet');
    bottomSheet.props.onClose();
    expect(mockOnCloseDatePicker).toHaveBeenCalledTimes(1);
  });

  it('should call onConfirmDate when confirm date button is pressed', () => {
    const container = containerInstance({
      showDatePicker: true,
      tempDate: new Date(),
    });
    const confirmButton = container.getByText('Confirmar');
    fireEvent.press(confirmButton);
    expect(mockOnConfirmDate).toHaveBeenCalledTimes(1);
  });

  it('should show transfer between wallets UI when transferBeetweenWallets is true', () => {
    const container = containerInstance({ transferBeetweenWallets: true });
    expect(container.getByText('Selecione as carteiras')).toBeTruthy();
    expect(container.getByText('De')).toBeTruthy();
    expect(container.getByText('Para')).toBeTruthy();
  });

  it('should show single wallet UI when transferBeetweenWallets is false', () => {
    const container = containerInstance({ transferBeetweenWallets: false });
    expect(container.getByText('Selecione a carteira')).toBeTruthy();
  });

  it('should call onSwapWallets when swap button is pressed', () => {
    const container = containerInstance({ transferBeetweenWallets: true });
    const swapButton = container.getByTestId('swap-wallets-button');
    fireEvent.press(swapButton);
    expect(mockOnSwapWallets).toHaveBeenCalledTimes(1);
  });
});
