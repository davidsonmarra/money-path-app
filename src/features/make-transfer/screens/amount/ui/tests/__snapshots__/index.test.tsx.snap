// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AmountContainer should render correctly 1`] = `
<RNCSafeAreaView
  edges={
    {
      "bottom": "additive",
      "left": "additive",
      "right": "additive",
      "top": "additive",
    }
  }
>
  <View
    style={
      {
        "alignItems": "center",
        "backgroundColor": "#2A5C99",
        "flexDirection": "row",
        "justifyContent": "space-between",
        "paddingBottom": 12,
        "paddingHorizontal": 16,
        "paddingTop": 12,
        "width": "100%",
      }
    }
  >
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "opacity": 1,
        }
      }
      testID="btn-left-icon"
    >
      <RNSVGSvgView
        align="xMidYMid"
        bbHeight={24}
        bbWidth={24}
        fill="#FFF"
        focusable={false}
        height={24}
        meetOrSlice={0}
        minX={0}
        minY={-960}
        style={
          [
            {
              "backgroundColor": "transparent",
              "borderWidth": 0,
            },
            {
              "flex": 0,
              "height": 24,
              "width": 24,
            },
          ]
        }
        testID="icon-back"
        vbHeight={960}
        vbWidth={960}
        width={24}
      >
        <RNSVGGroup
          fill={
            {
              "payload": 4294967295,
              "type": 0,
            }
          }
          propList={
            [
              "fill",
            ]
          }
        >
          <RNSVGPath
            d="M360-240 120-480l240-240 56 56-144 144h568v80H272l144 144z"
            fill={
              {
                "payload": 4278190080,
                "type": 0,
              }
            }
          />
        </RNSVGGroup>
      </RNSVGSvgView>
    </View>
    <Text
      style={
        [
          {
            "color": "#D9D9D9",
            "fontSize": 16,
            "fontWeight": 600,
            "lineHeight": 16,
          },
          {
            "color": "#FFF",
          },
        ]
      }
    >
      Valor da transferência
    </Text>
    <View />
  </View>
  <View
    style={
      {
        "backgroundColor": "#232E38",
      }
    }
  >
    <View
      style={
        {
          "backgroundColor": "#2A5C99",
          "paddingHorizontal": 24,
          "paddingVertical": 16,
          "width": "100%",
        }
      }
    >
      <TextInput
        keyboardType="numeric"
        onBlur={[Function]}
        onChangeText={[Function]}
        onFocus={[Function]}
        placeholder="0,00"
        placeholderTextColor="#FFFFFF65"
        selectTextOnFocus={true}
        style={
          {
            "color": "#FFF",
            "fontSize": 40,
            "fontWeight": "700",
            "textAlign": "right",
            "width": "100%",
          }
        }
        testID="input-amount"
        value="R$ 0,00"
      />
    </View>
  </View>
  <RNCSafeAreaView
    edges={
      {
        "bottom": "additive",
        "left": "off",
        "right": "off",
        "top": "off",
      }
    }
    style={
      {
        "backgroundColor": "#232E38",
        "flex": 1,
        "justifyContent": "space-between",
        "paddingHorizontal": 24,
        "paddingVertical": 16,
      }
    }
  >
    <View>
      <View>
        <Text
          style={
            [
              {
                "color": "#D9D9D9",
                "fontSize": 20,
                "fontWeight": 600,
                "lineHeight": 24,
              },
              undefined,
            ]
          }
        >
          Selecione a carteira
        </Text>
        <View
          style={
            {
              "height": 12,
              "width": 0,
            }
          }
          testID="spacer"
        />
        <Text
          style={
            [
              [
                {
                  "color": "#D9D9D9",
                  "fontSize": 14,
                  "lineHeight": 20,
                },
                {
                  "fontWeight": 400,
                },
              ],
              undefined,
            ]
          }
        >
          Para criar uma transferência, precisamos que você informe a carteira:
        </Text>
        <View
          style={
            {
              "height": 12,
              "width": 0,
            }
          }
          testID="spacer"
        />
        <View
          style={
            {
              "alignItems": "center",
              "flexDirection": "row",
              "justifyContent": "space-between",
              "marginHorizontal": "auto",
            }
          }
        >
          <View
            accessibilityState={
              {
                "busy": undefined,
                "checked": undefined,
                "disabled": false,
                "expanded": undefined,
                "selected": undefined,
              }
            }
            accessibilityValue={
              {
                "max": undefined,
                "min": undefined,
                "now": undefined,
                "text": undefined,
              }
            }
            accessible={true}
            collapsable={false}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              {
                "alignItems": "center",
                "backgroundColor": "#232E38",
                "borderColor": "#FFFFFF33",
                "borderRadius": 8,
                "borderWidth": 1,
                "flexDirection": "row",
                "gap": 16,
                "justifyContent": "space-between",
                "minHeight": 48,
                "opacity": 1,
                "padding": 12,
              }
            }
            testID="source-wallet-dropdown"
          >
            <View
              style={
                {
                  "alignItems": "center",
                  "backgroundColor": "#FF0000",
                  "borderRadius": 50,
                  "height": 48,
                  "justifyContent": "center",
                  "width": 48,
                }
              }
            >
              <RNSVGSvgView
                align="xMidYMid"
                bbHeight={32}
                bbWidth={32}
                fill="#D9D9D9"
                focusable={false}
                height={32}
                meetOrSlice={0}
                minX={0}
                minY={-960}
                style={
                  [
                    {
                      "backgroundColor": "transparent",
                      "borderWidth": 0,
                    },
                    {
                      "flex": 0,
                      "height": 32,
                      "width": 32,
                    },
                  ]
                }
                testID="icon-wallet"
                vbHeight={960}
                vbWidth={960}
                width={32}
              >
                <RNSVGGroup
                  fill={
                    {
                      "payload": 4292467161,
                      "type": 0,
                    }
                  }
                  propList={
                    [
                      "fill",
                    ]
                  }
                >
                  <RNSVGPath
                    d="M240-160q-66 0-113-47T80-320v-320q0-66 47-113t113-47h480q66 0 113 47t47 113v320q0 66-47 113t-113 47zm0-480h480q22 0 42 5t38 16v-21q0-33-23.5-56.5T720-720H240q-33 0-56.5 23.5T160-640v21q18-11 38-16t42-5m-74 130 445 108q9 2 18 0t17-8l139-116q-11-15-28-24.5t-37-9.5H240q-26 0-45.5 13.5T166-510"
                    fill={
                      {
                        "payload": 4278190080,
                        "type": 0,
                      }
                    }
                  />
                </RNSVGGroup>
              </RNSVGSvgView>
            </View>
            <RNSVGSvgView
              align="xMidYMid"
              bbHeight={20}
              bbWidth={20}
              fill="#D9D9D9"
              focusable={false}
              height={20}
              meetOrSlice={0}
              minX={0}
              minY={-960}
              style={
                [
                  {
                    "backgroundColor": "transparent",
                    "borderWidth": 0,
                  },
                  {
                    "flex": 0,
                    "height": 20,
                    "width": 20,
                  },
                ]
              }
              testID="icon-chevron-down"
              vbHeight={960}
              vbWidth={960}
              width={20}
            >
              <RNSVGGroup
                fill={
                  {
                    "payload": 4292467161,
                    "type": 0,
                  }
                }
                propList={
                  [
                    "fill",
                  ]
                }
              >
                <RNSVGPath
                  d="M480-345 240-585l56-56 184 184 184-184 56 56-240 240z"
                  fill={
                    {
                      "payload": 4278190080,
                      "type": 0,
                    }
                  }
                />
              </RNSVGGroup>
            </RNSVGSvgView>
          </View>
        </View>
      </View>
      <View
        style={
          {
            "alignItems": "center",
            "paddingHorizontal": 24,
            "paddingVertical": 16,
          }
        }
      >
        <Text
          style={
            [
              [
                {
                  "color": "#D9D9D9",
                  "fontSize": 14,
                  "lineHeight": 20,
                },
                {
                  "fontWeight": 400,
                },
              ],
              undefined,
            ]
          }
        >
          Data e hora da transferência
        </Text>
        <View
          style={
            {
              "height": 8,
              "width": 0,
            }
          }
          testID="spacer"
        />
        <View
          accessibilityState={
            {
              "busy": undefined,
              "checked": undefined,
              "disabled": undefined,
              "expanded": undefined,
              "selected": undefined,
            }
          }
          accessibilityValue={
            {
              "max": undefined,
              "min": undefined,
              "now": undefined,
              "text": undefined,
            }
          }
          accessible={true}
          collapsable={false}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            {
              "alignItems": "center",
              "backgroundColor": "#2A5C99",
              "borderRadius": 8,
              "justifyContent": "center",
              "opacity": 1,
              "padding": 20,
            }
          }
          testID="date-button"
        >
          <Text
            style={
              [
                {
                  "color": "#D9D9D9",
                  "fontSize": 20,
                  "fontWeight": 600,
                  "lineHeight": 24,
                },
                {
                  "color": "#FFF",
                  "fontSize": 32,
                  "lineHeight": 40,
                },
              ]
            }
          >
            25
          </Text>
          <Text
            style={
              [
                [
                  {
                    "color": "#D9D9D9",
                    "fontSize": 14,
                    "lineHeight": 20,
                  },
                  {
                    "fontWeight": 400,
                  },
                ],
                {
                  "color": "#FFF",
                  "fontSize": 20,
                },
              ]
            }
          >
            de
             
            julho
          </Text>
          <View
            style={
              {
                "height": 8,
                "width": 0,
              }
            }
            testID="spacer"
          />
          <Text
            style={
              [
                [
                  {
                    "color": "#D9D9D9",
                    "fontSize": 14,
                    "lineHeight": 20,
                  },
                  {
                    "fontWeight": 400,
                  },
                ],
                {
                  "color": "#FFF",
                  "fontSize": 16,
                },
              ]
            }
          >
            15:07
          </Text>
        </View>
      </View>
    </View>
    <View>
      <View
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": false,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "alignItems": "center",
            "backgroundColor": "#EAC43D",
            "borderRadius": 12,
            "flexDirection": "row",
            "height": 48,
            "justifyContent": "space-between",
            "opacity": 1,
            "paddingHorizontal": 24,
            "paddingVertical": 12,
            "width": "100%",
          }
        }
        testID="btn-confirm"
      >
        <View />
        <Text
          style={
            [
              [
                {
                  "color": "#D9D9D9",
                  "fontSize": 16,
                  "lineHeight": 24,
                },
                {
                  "fontWeight": 600,
                },
              ],
              {
                "color": "#FFF",
              },
            ]
          }
        >
          Continuar
        </Text>
        <View />
      </View>
    </View>
  </RNCSafeAreaView>
</RNCSafeAreaView>
`;
